svm/api.go

```diff
@@ -0,0 +1,51 @@
+package svm
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/api/config"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/common/util"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/svm/state"
+)
+
+// SVM is an entry point for all SVM operations.
+type SVM struct {
+	state *state.TransactionProcessor
+	log   log.Logger
+}
+
+// New creates a new `SVM` instance from the given `state` and `logger`.
+func New(state *state.TransactionProcessor, logger log.Log) *SVM {
+	return &SVM{state, log.NewDefault("svm")}
+}
+
+// SetupGenesis creates new accounts and adds balances as dictated by `conf`.
+func (svm *SVM) SetupGenesis(conf *config.GenesisConfig) error {
+	if conf == nil {
+		conf = config.DefaultGenesisConfig()
+	}
```

Comment:
This will replace the pointer, not what it points to - this check needs to be done somewhere else or differently somehow.

Reply:
Why do we need to replace the value `conf` initially points to?

- Receive a pointer to a `config.GenesisConfig`.
- If invalid (`nil`), we point instead to the default configuration options.
- By now, our pointer will point to either the given configuration options or the default ones.
- We're ready to read the value pointed by `config` and use it for initialization.

Reply:
It's fine for this local function - the value of `conf` will be correct. However, the previous implementation updated the global configuration (`app.Config.Genesis`) to the default if it's `nil`.

I'm not sure this will have any impact, as this may not be read anywhere other than in this function. If that's the case (please make sure) you can leave it the way it is now.

Reply:
I checked and we don't access `.Genesis` anywhere other than this function - so you can leave this as-is.

Reply:
Okay, I didn't realize the old version updated the value as well, I guess that's where my confusion was.